### PR TITLE
Use EcosystemBackend

### PR DESCRIPTION
**Pull request by humaton**
implements: openshift.io/issues/3698

### CODE REVIEW

File: f8a_worker/dispatcher/foreach.py [hunk 1]
```diff
@@ -112,7 +117,7 @@ def iter_unknown_dependencies(storage_pool, node_args):
         for element in aggregated["result"]:
             epv = element.split(':')
             ecosystem = epv[0]
-            if ecosystem == 'maven':
+            if ecosystem == EcosystemBackend.maven:
```
Comment (msrb) [inline f8a_worker/dispatcher/foreach.py]: This won't work (?) Or am I missing something here?

**Comment by humaton** [comment]
[TEST]

**Comment by msrb** [comment]
This also needs to be fixed: https://github.com/fabric8-analytics/fabric8-analytics-worker/blob/master/f8a_worker/defaults.py#L159
We need ecosystem backend there.

**Comment by msrb** [comment]
@humaton have you tested this?

**Comment by msrb** [comment]
[test]

**Pull request merged**